src/P3Scheme.jl

```diff
@@ -0,0 +1,149 @@
+"""
+Predicted particle properties scheme (P3) for ice, which includes:
+ - m(D) regime
+ - A(D) regime
+ - particle fall speed regime
+ - add more (source/sink) as we go! (TODO)
+"""
+module P3Scheme
+
+import CloudMicrophysics as CM
+import Thermodynamics as TD
+import NonlinearSolve as NLS
+
+const CT = CM.CommonTypes
+const CO = CM.Common
+const CMP = CM.Parameters
+const APS = CMP.AbstractCloudMicrophysicsParameters
+
+const FT = Float64
+
+const ρ_i::FT = 917.0
+const β_va::FT = 1.9
+const α_va::FT = (7.38e-11) * 10^((6 * β_va) - 3)
+const D_th::FT = ((FT(π) * ρ_i) / (6 * α_va))^(1 / (β_va - 3))
+const γ::FT = 0.2285
+const σ::FT = 1.88
+const δ_o::FT = 5.83 # UNSURE about these four constants below
+const C_o::FT = 0.6
+const a_o::FT = 1.7e-3
+const b_o::FT = 0.8
+
+"""
+breakpoints(ρ_r, F_r)
+
+- ρ_r: predicted rime density (q_rim/B_rim)
+- F_r: rime mass fraction (q_rim/q_i)
+
```

Comment:
Could you define what are the units of the input and output? And also what is D_cr, D_gr, ρ_g, ρ_d?